airbyte-integrations/connectors/source-google-sheets/google_sheets_source/google_sheets_source.py

```diff
@@ -100,16 +100,18 @@ def read(
         # For each sheet in the spreadsheet, get a batch of rows, and as long as there hasn't been
         # a blank row, emit the row batch
         sheet_to_column_index_to_name = Helpers.get_available_sheets_to_column_index_to_name(client, spreadsheet_id, sheet_to_column_name)
+        sheet_parameters = Helpers.get_sheets_properties(client, spreadsheet_id)
         for sheet in sheet_to_column_index_to_name.keys():
             logger.info(f"Syncing sheet {sheet}")
             column_index_to_name = sheet_to_column_index_to_name[sheet]
             row_cursor = 2  # we start syncing past the header row
-            while True:
+            while row_cursor <= sheet_parameters[sheet]:
```

Comment:
@yevhenii-ldv wouldn't this overcount in the case that `row_cusror + batchsize` is greater than `sheet_parameters[sheet]`?

Reply:
@sherifnada It is necessary that the initial row exists, if the final row of the interval will go beyond the table - that's okay, we will only return the real data of the table and in the next iteration will exit the loop.